exchanges/bybit/bybit_test.go

```diff
@@ -0,0 +1,3303 @@
+)
+
+var b Bybit
+
+func TestMain(m *testing.M) {
+	b.SetDefaults()
+	cfg := config.GetConfig()
+	err := cfg.LoadConfig("../../testdata/configtest.json", true)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	exchCfg, err := cfg.GetExchangeConfig("Bybit")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	exchCfg.API.AuthenticatedSupport = true
+	exchCfg.API.AuthenticatedWebsocketSupport = false
+	exchCfg.API.Credentials.Key = apiKey
+	exchCfg.API.Credentials.Secret = apiSecret
+	b.Websocket = sharedtestvalues.NewTestWebsocket()
+	err = b.Setup(exchCfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func areTestAPIKeysSet() bool {
+	return b.ValidateAPICredentials(b.GetDefaultCredentials()) == nil
+}
+
+func TestStart(t *testing.T) {
+	t.Parallel()
+	err := b.Start(nil)
+	if !errors.Is(err, common.ErrNilPointer) {
+		t.Fatalf("received: '%v' but expected: '%v'", err, common.ErrNilPointer)
+	}
+	var testWg sync.WaitGroup
+	err = b.Start(&testWg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	testWg.Wait()
+}
+
+// test cases for SPOT
+
+func TestGetAllSpotPairs(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetAllSpotPairs(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetOrderBook(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetOrderBook(context.Background(), "BTCUSDT", 100)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetMergedOrderBook(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetMergedOrderBook(context.Background(), "BTCUSDT", 2, 100)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetTrades(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetTrades(context.Background(), "BTCUSDT", 100)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetKlines(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetKlines(context.Background(), "BTCUSDT", "5m", 2000, time.Now().Add(-time.Hour*1), time.Now())
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGet24HrsChange(t *testing.T) {
+	t.Parallel()
+	_, err := b.Get24HrsChange(context.Background(), "BTCUSDT")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = b.Get24HrsChange(context.Background(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetLastTradedPrice(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetLastTradedPrice(context.Background(), "BTCUSDT")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = b.GetLastTradedPrice(context.Background(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetBestBidAskPrice(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetBestBidAskPrice(context.Background(), "BTCUSDT")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = b.GetBestBidAskPrice(context.Background(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCreatePostOrder(t *testing.T) {
+	t.Parallel()
+
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+	_, err := b.CreatePostOrder(context.Background(), &PlaceOrderRequest{
```

Comment:
This executes an order but it fails on unmarshalling
```
{"ret_code":0,"ret_msg":"","ext_code":null,"ext_info":null,"result":{"accountId":"21937254","symbol":"XRPUSDT","symbolName":"XRPUSDT","orderLinkId":"1656548951565541","orderId":"1189613260062690304","transactTime":"1656548951572","price":"0","origQty":"10","executedQty":"0","status":"NEW","timeInForce":"GTC","type":"MARKET","side":"SELL"}}
```
```json
bybit_test.go:185: json: cannot unmarshal string into Go struct field PlaceOrderResponse.result.transactTime of type int64
```

Reply:
```json
json: cannot unmarshal string into Go struct field PlaceOrderResponse.result.accountId of type int64
json: cannot unmarshal string into Go struct field PlaceOrderResponse.result.orderId of type int64
```

Reply:
Sorry for missing them out. I have got it fixed now. @shazbert